File: Atividade_12/Produto.go
package main

import "fmt"

type Produto struct {
	nome  string
	preco float64
}

func (p Produto) SomarPreco(outro Produto) float64 {
	return p.preco + outro.preco
}

func main() {
	produto1 := Produto{"Produto 1", 30}
	produto2 := Produto{"Produto 2", 45}

	fmt.Printf("Soma dos preços: R$ %.2f\n", produto1.SomarPreco(produto2))
}

File: Atividade_12/Produto.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public double somarPreco(Produto outro) {
        return this.preco + outro.preco;
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto 1", 30);
        Produto produto2 = new Produto("Produto 2", 45);

        System.out.println("Soma dos preços: R$ " + produto1.somarPreco(produto2));
    }
}
